### ISSUE TITLE
CosmosDB scraper returns 'Nullable object must have a value' exception 

### ISSUE THREAD

**Issue by ricardoalcazar** [tomkerkhove/promitor#789]
## Expected Behavior
Query CosmosDB TotalRequests and ProvisionedThroughput by database account name.

## Actual Behavior
Promitor container starts successfully, but fails the scraping process with the following exception:

```
Hosting environment: Production
Content root path: /app
Now listening on: http://[::]:80
Application started. Press Ctrl+C to shut down.
info: Runtime[0]
      [2019-12-03 19:58:02Z] Scraping Azure Monitor - 12/03/2019 19:58:02 +00:00
info: Runtime[0]
      [2019-12-03 19:58:02Z] Metrics declaration is using spec version v1
info: Runtime[0]
      [2019-12-03 19:58:02Z] Scraping 'azure_cosmos_db_provisioned_throughput' for resource type 'CosmosDb'
fail: Runtime[0]
      [2019-12-03 19:58:03Z] Nullable object must have a value.
System.InvalidOperationException: Nullable object must have a value.
   at Promitor.Core.Scraping.Scraper`1.ScrapeAsync(ScrapeDefinition`1 scrapeDefinition)
```

## Steps to Reproduce the Problem

  1. Configure Service Principal with Azure Monitor Reader permission
  2. Configure metrics & runtime configuration per documentation for CosmosDB
  3. Start container using env-var PROMITOR_AUTH_APPID and PROMITOR_AUTH_APPKEY

## Configuration
Provide insights in the configuration that you are using:
- Configured scraping schedule: 

<details>
<summary>Used scraping configuration</summary>

```yaml
version: v1
azureMetadata:
  tenantId: *OMITTED*
  subscriptionId: *OMITTED*
  resourceGroupName: DV-COSMOSDB-RG
metricDefaults:
  scraping:
    schedule: "0 * * ? * *" # every minute
metrics:
  - name: azure_cosmos_db_provisioned_throughput
    description: "Demo cosmos query"
    resourceType: CosmosDb
    azureMetricConfiguration:
      metricName: TotalRequests
      aggregation:
        type: Count
    resources:
    - dbName: dev-cdb
```
  
</details>

## Specifications

  - Version: 1
  - Platform: Ubuntu
  - Subsystem:

**Comment by tomkerkhove** [comment]
Thanks for the issue, can you have another look at the version as it says 18 while we don't have that version 🤔

**Comment by tomkerkhove** [comment]
Hm it looks like I cannot reproduce it and output is as expected:
```yaml
# HELP azure_cosmos_db_provisioned_throughput Demo cosmos query
# TYPE azure_cosmos_db_provisioned_throughput gauge
azure_cosmos_db_provisioned_throughput{resource_group="promitor-sources",subscription_id="0f9d7fea-99e8-4768-8672-06a28514f77e",resource_uri="subscriptions/0f9d7fea-99e8-4768-8672-06a28514f77e/resourceGroups/promitor-sources/providers/Microsoft.DocumentDB/databaseAccounts/promitor",instance_name="promitor"} 400 1575446705223
# HELP azure_cosmos_db_total_requests Demo cosmos query
# TYPE azure_cosmos_db_total_requests gauge
azure_cosmos_db_total_requests{resource_group="promitor-sources",subscription_id="0f9d7fea-99e8-4768-8672-06a28514f77e",resource_uri="subscriptions/0f9d7fea-99e8-4768-8672-06a28514f77e/resourceGroups/promitor-sources/providers/Microsoft.DocumentDB/databaseAccounts/promitor",instance_name="promitor"} 0 1575446705099
```

Based on this config:
```yaml
version: v1
azureMetadata:
  tenantId: XYZ
  subscriptionId: ABC
  resourceGroupName: promitor
metricDefaults:
  aggregation:
    interval: 00:05:00
  scraping:
    # Every minute
    schedule: "0 * * ? * *"
metrics:
  - name: azure_cosmos_db_total_requests
    description: "Demo cosmos query"
    resourceType: CosmosDb
    azureMetricConfiguration:
      metricName: TotalRequests
      aggregation:
        type: Count
    resources:
    - dbName: promitor
      resourceGroupName: promitor-sources
  - name: azure_cosmos_db_provisioned_throughput
    description: "Demo cosmos query"
    resourceType: CosmosDb
    azureMetricConfiguration:
      metricName: ProvisionedThroughput
      aggregation:
        type: Maximum
```

**Comment by tomkerkhove** [comment]
Reproduced.

You don't specify `metricDefaults.aggregation.interval` nor `[metric].azureMetricConfiguration.aggregation.interval` so we don't know what aggregation to use.

I'd recommend to define `metricDefaults.aggregation.interval` as a default, see: https://promitor.io/configuration/v1.x/metrics/

I've created #791 to improve our validation for this, ok for you if I close this issue since it's more of a config/validation issue for that spec?
